packages/context/src/context.ts

```diff
@@ -349,6 +355,120 @@ export class Context extends EventEmitter {
+   *
+   * @param key The key for the binding to be configured
+   */
+  configure<ConfigValueType = BoundValue>(
+    key: BindingAddress = '',
+  ): Binding<ConfigValueType> {
+    const keyForConfig = BindingKey.buildKeyForConfig(key);
+    const bindingForConfig = this.bind<ConfigValueType>(keyForConfig).tag({
+      config: key,
+    });
+    return bindingForConfig;
+  }
+
+  /**
+   * Get the value or promise of configuration for a given binding by key
+   *
+   * @param key Binding key
+   * @param configPath Property path for the option. For example, `x.y`
+   * requests for `<config>.x.y`. If not set, the `<config>` object will be
+   * returned.
+   * @param resolutionOptions Options for the resolution.
+   * - optional: if not set or set to `true`, `undefined` will be returned if
+   * no corresponding value is found. Otherwise, an error will be thrown.
+   */
+  getConfigAsValueOrPromise<ConfigValueType>(
+    key: BindingAddress,
+    configPath?: string,
+    resolutionOptions?: ResolutionOptions,
+  ): ValueOrPromise<ConfigValueType | undefined> {
+    this.setupConfigurationResolverIfNeeded();
+    return this.configResolver.getConfigAsValueOrPromise(
+      key,
+      configPath,
+      resolutionOptions,
+    );
+  }
+
+  /**
+   * Set up the configuration resolver if needed
+   */
+  protected setupConfigurationResolverIfNeeded() {
+    if (!this.configResolver) {
+      // First try the bound ConfigurationResolver to this context
+      const configResolver = this.getSync<ConfigurationResolver>(
+        `${BindingKey.CONFIG_NAMESPACE}.resolver`,
+        {
+          optional: true,
+        },
+      );
+      if (configResolver) {
+        this.configResolver = configResolver;
+      } else {
+        // Fallback to DefaultConfigurationResolver
+        this.configResolver = new DefaultConfigurationResolver(this);
+      }
+    }
+    return this.configResolver;
+  }
+
+  /**
+   * Resolve configuration for the binding by key
+   *
+   * @param key Binding key
+   * @param configPath Property path for the option. For example, `x.y`
+   * requests for `<config>.x.y`. If not set, the `<config>` object will be
+   * returned.
+   * @param resolutionOptions Options for the resolution.
+   */
+  async getConfig<ConfigValueType>(
+    key: BindingAddress,
+    configPath?: string,
+    resolutionOptions?: ResolutionOptions,
+  ): Promise<ConfigValueType | undefined> {
+    return await this.getConfigAsValueOrPromise<ConfigValueType>(
+      key,
+      configPath,
+      resolutionOptions,
+    );
+  }
+
+  /**
+   * Resolve configuration synchronously for the binding by key
+   *
+   * @param key Binding key
+   * @param configPath Property path for the option. For example, `x.y`
+   * requests for `config.x.y`. If not set, the `config` object will be
+   * returned.
+   * @param resolutionOptions Options for the resolution.
+   */
+  getConfigSync<ConfigValueType>(
```

Comment:
Is it a good idea to introduce the sync version? I mean what are the use cases where `await ctx.getConfig()` cannot be used?

Reply:
> Is it a good idea to introduce the sync version? I mean what are the use cases where `await ctx.getConfig()` cannot be used?

☝️